### ISSUE TITLE
Unable to sync the blockchain when no public IP is available

### ISSUE THREAD

**Issue by mckelvin** [NebulousLabs/Sia#2525]
Bug Report
==========

**Stack Trace or error message**

I'm using Sia in an environment where public IP is not available (my ISP doesn't assign a public IP for me). The problem is: The Sia doesn't sync automatically:

```
 $ siac
Synced: No
Height: 128116
Progress (estimated): 92.3%

$ siac gateway list
No peers to show.
```

When I tried to connect a remote gateway peer manually

```
$ siac gateway connect X.X.X.X:9981 (Which is an available gateway)
Could not add peer: peer rejected our header: claimed hostname (192.168.1.100, which is my local IP) does not match conn.RemoteAddr (Y.Y.Y.Y, which is the the public IP of my ISP)
```

But [some of the peers](https://www.reddit.com/r/siacoin/comments/7iga3x/sia_wallet_wont_start_sync/) will accept me:

```
$ siac gateway connect 104.168.152.120:9981
Added 104.168.152.120:9981 to peer list.
```

I don't know why majority of the peers reject me just because of the special network setting. As a Sia user, I'd like to be able to sync automatically the blockchain even when in the special case. 

**Environment**
* Sia version: 1.3.0
* OS: macOS

**Comment by lukechampine** [comment]
We removed the "claimed hostname" in the upcoming v1.3.1 release. Unfortunately, it may take some time for nodes to upgrade. Until then, you will have difficulty connecting to some nodes. Your best option is to keep a list of peers that you successfully connected to, so that you can reconnect to them if needed.

**Comment by mckelvin** [comment]
@lukechampine Thanks for your reply. I just compiled the latest version from source. Now the claimed hostname is no longer an intranet IP (~192.168.X.X~), now it's a public IP address (i.e.: `PUBLIC_IP_A` as shown below).


```
$ siac gateway connect 107.199.252.74:9981
Could not add peer: peer rejected our header: claimed hostname (PUBLIC_IP_A:9981) does not match conn.RemoteAddr (PUBLIC_IP_B:9981)

$ siac gateway
Address: PUBLIC_IP_C:9981
Active peers: 0

$ siac version
Sia Client v1.3.1
Sia Daemon v1.3.1
```

The remote gateway IP (`107.199.252.74:9981`) comes from https://siahub.info/ , which is using `1.3.0`

But the problem is still there. You may doubt why there're 3 different public address. Actually when I visit different websites, different websites see different IP of mine. None of them are owned by me privately. I guess there's a public IP pool of the ISP. When I visit server, different public IP is used.

* * *

I just started the latest version of daemon a few minutes ago, right now I can see it can sync up automatically. Some remote gateways with version 1.3.1 can be found in `siac gateway list`. So for me. I think the issue is resolved by upgrading to `1.3.1` .

Thanks again for your help.
